cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QWidget *lpTempWidget = new QWidget();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerSavePath();
+
+    // Center the popup(s) that are to come in the near future
+    lpTempWidget->move(QApplication::desktop()->screen()->rect().center() - lpTempWidget->rect().center());
+    
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        QMessageBox::critical(lpTempWidget, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+    }
+    else
+    {
+        int lnStatusCode = lpNetworkReply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+        if (lnStatusCode == 301 || lnStatusCode == 302)
+        {
+            // Was a redirect, will need to start over!
+            QUrl lcRedirectUrl = lpNetworkReply->attribute(QNetworkRequest::RedirectionTargetAttribute).toUrl();
+            downloadSpoilersFile(lcRedirectUrl);
+        }
+        else
+        {
+            // Read downloaded file into buffer
+            mcDownloadedFileData = lpNetworkReply->readAll();
+
+            if (saveSpoilersToFile(lsSpoilerSavePath))
+            {
+                // Reload the card database to import the spoilers
+                QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+                QMessageBox::information(lpTempWidget, tr("Success"), tr("Spoiler database updated and saved successfully to\n%1").arg(lsSpoilerSavePath));
+            }
+            else
+            {
+                QMessageBox::critical(lpTempWidget, tr("Error"), tr("Spoiler database updated but could not be saved to %1").arg(lsSpoilerSavePath));
+            }
+        }
+}
+
+lpNetworkReply->deleteLater();
+lpTempWidget->deleteLater();
+}
+
+void SpoilerBackgroundUpdaterThread::actSeeIfSpoilerSeasonIsActive()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QWidget *lpTempWidget = new QWidget();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerSavePath();
+
+    // Center the popup(s) that are to come in the near future
+    lpTempWidget->move(QApplication::desktop()->screen()->rect().center() - lpTempWidget->rect().center());
+
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        // If there's a network issue
+        mbIsActiveSpoilerSeason = false;
+        QMessageBox::critical(lpTempWidget, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+    }
+    else
+    {
+        // If the file reads "enabled" then we're in spoiler season (true)
+        // If the file reads anything else then we're NOT in spoiler season (false)
+        QByteArray laFileData = lpNetworkReply->readAll();
+        mbIsActiveSpoilerSeason = QString(laFileData).contains("enabled", Qt::CaseInsensitive);
+    }
```

Comment:
To avoid the hack code at lines 135-140, you could start the timer here instead of in the constructor.